### ISSUE TITLE
Please fix the referencing issues flagged by VS

### ISSUE THREAD

**Issue by TehWardy** [aspnet/AspNetCore#9022]
### Describe the bug
This appears to be a problem throughout the .Net Core ecosystem. 
every part of .Net Core explicitly depends on older versions of assemblies instead of just having a minimum version resulting in me getting a lot of build warnings for simply being up to date with my project references.

### To Reproduce
1. Using version 2.* of ASP.NET Core 
2. Create a new project, and add references to AspNet Core, the Azure Fluent API, and Newtonsoft.Json (as an example, these are pretty common references that people will have)
3. Complile the project
4. Witness the warning spam all over VS Error List window.

### Expected behavior
No warnings, this sort of thing should just be acceptable by default

### Additional context
Some examples I currently have to deal with ...
Microsoft.AspNetCore.App v2.1.1. requires Microsoft.AspNetCore.Razor.Design >=2.1.1 && <2.2.0
Micorsoft.NET.Sdk.Functions requires NewtonSoft.Json v9.0.1 
Found conflicts between versions Microsoft.Extensions.Primitives
Found Conflicts between versions of System.Net.Http.Formatting

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @TehWardy.
We do plan to address the issues you'd get when creating a new project from one of our default in 3.0 release. Some of that work is tracked by https://github.com/aspnet/AspNetCore/issues/5442.
As for the issues you get from referencing other packages, those are valid warnings and I'm not sure it makes sense for us to hide those.

@rynowak, any other thoughts?

**Comment by mkArtakMSFT** [comment]
Also, make sure you reference compatible packages, as 2.1 and 2.2 are not compatible and you'd get valid warnings in case you reference packages targeting different frameworks.

**Comment by TehWardy** [comment]
The point was not that I "can" reference different versions of packages it was issues like the latest versions of Microsoft's packages just don't work / are not compatible with each other by default or force older dependencies.

Example:
Why force a dependency of version 9.0.1 of Newtonsoft.Json in a brand new technology like Azure functions ... that means all the code i package up in a function now cannot be up to date with that package and since i'm packaging up some of my core assemblies you're forcing my whole code base to be 3 major versions out of date and for what reason?
v9 ... really? you guys couldn't just support a min version of 9, it had to be specifically that version?

Simply put ... the latest version of Microsoft's web stack should be compatible with the latest version of its own web stack BY DEFAULT and it simply isn't, and the problem gets worse when I want to use multiple parts of MS Nuget offerings that simply aren't compatible because they require different versions of their dependencies which simply can't be resolved.

This is poor product planning and testing process at it's worst and it creates problems that go beyond your code base.
All i'm asking is that MS at least be consistent across it's packages to it's easier to determine a working set that all click together instead forcing "DLL hell" in to my code base.
